### PR TITLE
Enable python console when debugging Python lambdas

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Related Issue(s)
#1165

## Screenshots (if appropriate)
Screen Shot 2021-02-03 at 4 28 31 PM

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt [hunk 1]
```diff
@@ -33,6 +37,9 @@ object PythonDebugUtils {
                 )
             )
 
+            val sdk = ProjectRootManager.getInstance(environment.project).projectSdk?.takeIf { it.sdkType is PythonSdkType }
+            state.consoleBuilder = PyDebugConsoleBuilder(environment.project, sdk)
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt:41]: What happens if sdk is null? Does this still work or does it throw?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt:41]: It works, just seems to be stdlib only?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt [hunk 2]
```diff
@@ -33,6 +37,9 @@ object PythonDebugUtils {
                 )
             )
 
+            val sdk = ProjectRootManager.getInstance(environment.project).projectSdk?.takeIf { it.sdkType is PythonSdkType }
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt:40]: can the module have a different sdk than the project?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt:40]: Ya in IntelliJ

Guess we need to find a way to get the Module down here if we can..

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt:40]: We dont really have a way to get the module that far down thats event remote clean :/

**Review by abrooksv — COMMENTED** [review]
c

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt (hunk 2 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonDebugUtils.kt:40]: Can you add a TODO in here for that?

**Review by hunterwerlla — APPROVED** [review]

**Comment by abrooksv** [comment]
This got broken by #2431, do not merge

**Comment by abrooksv** [comment]
Fixed, please re-review @hunterwerlla

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt [hunk 3]
```diff
@@ -95,4 +99,8 @@ class SamRunningState(
         eventFile.writeText(settings.input)
         return eventFile.absolutePath
     }
+
+    override fun createConsole(executor: Executor): ConsoleView? = runBlocking(getCoroutineUiContext(disposable = environment)) {
+        super.createConsole(executor)
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt:104]: comment on the fact that certain consoles need to be created on EDT?
(though at this point we've probably switched between on and off EDT at least 3 times by now)

**Review by rli — APPROVED** [review]

**Review by hunterwerlla — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt (hunk 3 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunningState.kt]: Added

**Pull request merged**